daft/runners/ray_runner.py

```diff
@@ -652,6 +653,8 @@ def _build_partitions(
         build_remote = (
             fanout_pipeline if isinstance(task.instructions[-1], FanoutInstruction) else single_partition_pipeline
         )
+        if isinstance(task.instructions[0], ScanWithTask):
```

Comment:
also check if `task.instructions` is non empty

Reply:
Hmm isn't that guaranteed? We're already indexing into `task.instructions` in the statement above this.

Reply:
Looks like we don't guarantee anywhere when building a `PartitionTask` that the `instructions` will be non-empty, and downstream processing of the `PartitionTask` allows for `instructions` to be empty, so I'll add a quick driveby fix in the runners for this.